gui_qt/app.py

```
@@ -8175,14 +8193,28 @@ def _on_save_config(self) -> bool:
                 apply_advanced_settings(config, complete_advanced_values)
 
+            # Validate every field before either settings file is written. This
+            # avoids persisting project flags when the UI reports "未保存".
+            from project_context_settings import save_project_context_settings
+
+            project_settings_path = save_project_context_settings(
+                self.state.get_game_root(),
+                project_context_flags,
+            )
+            self._append_log(
+                f"当前项目上下文开关已保存：{project_settings_path}"
```

Ruff (0.15.20)

[warning] String contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF001)